app/models/user.rb

```diff
@@ -110,6 +112,18 @@ def validate
   before_validation :dummy_password_for_external_auth
   before_destroy :destroy_subscribed_widget_sets
 
+  def check_reference
+    present_ref = []
+    %w(miq_requests vms miq_widgets miq_templates).each do |association|
+      present_ref << association.classify unless public_send(association).first.nil?
+    end
+
+    unless present_ref.empty?
+      errors.add(:base, "user with id [#{id}] has references to other models: #{present_ref.join(" ")}")
```

Comment:
We should probably also mention the `userid` or `name`

Reply:
👍